Model/ProductProtection.php

```diff
@@ -631,4 +647,59 @@ private function addAdditionalOptions($item, $productId, $term, $leadToken, $lea
             'value' => $this->serializer->serialize($additionalOptions),
         ]);
     }
+
+    /**
+     * This function adds options unique to the Product Protection product as well as additional options customary to Magento
+     *
+     * @param $cartItemId
+     * @param $leadToken
+     * @param $quantity
+     * @param $product
+     * @param $item
+     * @param $term
+     * @param $productId
+     * @param $planId
+     * @param $listPrice
+     * @param $orderOfferPlanId
+     * @param $coverageType
+     * @return void
+     * @throws LocalizedException
+     */
+    private function processOptions(
+        $cartItemId,
+        $leadToken,
+        $quantity,
+        $product,
+        $item,
+        $term,
+        $productId,
+        $planId,
+        $listPrice,
+        $orderOfferPlanId,
+        $coverageType
+    ) {
+        $optionValues = [
+            self::PLAN_TYPE_CODE => $coverageType,
+            self::TERM_CODE => $term,
+            self::LIST_PRICE_CODE => $listPrice,
+            self::OFFER_PLAN_ID_CODE => $orderOfferPlanId,
+            self::LEAD_TOKEN => $leadToken,
+            self::ASSOCIATED_PRODUCT_SKU_CODE => $productId,
+            self::PLAN_ID_CODE => $planId,
+        ];
+
+        $leadQuantity = null;
+
+        if (isset($leadToken) && !isset($cartItemId)) {
+            $optionValues[self::LEAD_QUANTITY_CODE] = $quantity;
+            $leadQuantity = $quantity;
+        }
+
+        $options = $this->createOptions($product, $item, $optionValues);
+        $item->setOptions($options);
+
+        if ((isset($leadToken) && isset($leadQuantity)) || isset($term) || isset($productId)) {
+            $this->addAdditionalOptions($item, $productId, $term, $leadToken, $leadQuantity);
```

Comment:
Is it possible/problematic here for there to be a lead token with no lead quantity to be added because there's a term or product? Or are we validating elsewhere?

Reply:
Earlier in the parent function, if there is a quantity of 0, and a cartItemId, this is interpreted at the product needs to be removed.  If that happens the parent function returns much earlier.  If we've come this far, we have ruled out a situation where there is a cartItemId and 0 quantity.  But I guess if we have a cartItemId and a leadToken, we're failing to pass that quantity through as leadQuantity on line 693. @adityaagashe1 what do you think?

Reply:
@jay-faust-extend, @adityaagashe1 said this is good the way it is.